common/lib/xmodule/xmodule/modulestore/split_mongo/split.py

```diff
@@ -3014,6 +3015,39 @@ def ensure_indexes(self):
         """
         self.db_connection.ensure_indexes()
 
+    def _get_block_usage_locator(self, course_key, block_key):
+        return BlockUsageLocator(course_key, block_key.type, block_key.id)
+
+    def _format_course_structure(self, course_key, structure, version):
+        # Convert the root key to a locator string
+        root = unicode(self._get_block_usage_locator(course_key, structure[u'root']))
+        blocks = {}
+
+        for key, block in structure[u'blocks'].iteritems():
+            locator = unicode(self._get_block_usage_locator(course_key, key))
+            fields = block.get(u'fields', {})
+            children = fields.get(u'children', [])
+            children = [unicode(self._get_block_usage_locator(course_key, child)) for child in children]
+
+            blocks[locator] = Block(locator,
+                                    block[u'block_type'],
+                                    display_name=fields.get(u'display_name', None),
+                                    format=fields.get(u'format', None),
+                                    graded=fields.get(u'graded', False),
+                                    children=children)
+
+        return CourseStructure(root, blocks, version=version)
+
+    def get_course_structure(self, course_id, version=None):
+        if not version:
+            course_index = self.get_course_index(course_id)
+            versions = course_index[u'versions']
+            version = versions.get(ModuleStoreEnum.BranchName.published) or versions.get(
+                ModuleStoreEnum.BranchName.draft)
+
+        structure = self.get_structure(course_id, version)
+        return self._format_course_structure(course_id, structure, version)
```

Comment:
The stored data in split doesn't have an inheritance computed on it, so you'd need to do that manually in split as well as in old-mong.